test/cases/function/table_func_unnest.test

```diff
@@ -0,0 +1,43 @@
+#Test for str case
+select * from unnest('{"a":1}') as f;
+select * from unnest('{"a":1}', '$.a') as f;
+select * from unnest('{"a":1}', '$.a', true) as f;
+select * from unnest('{"a":1}');
+select * from unnest('{"a":1}', '$.a');
+select * from unnest('{"a":1}', '$.a', true);
+
+#Test for col case
+drop table if exists t1;
+create table t1 (a json,b int);
+insert into t1 values ('{"a":1,"b":[{"c":2,"d":3},false,4],"e":{"f":true,"g":[null,true,1.1]}}',1);
+insert into t1 values ('[1,true,false,null,"aaa",1.1,{"t":false}]',2);
+select * from t1;
+select * from unnest(t1.a) as f;
+select * from unnest(t1.a, "$.b") as f;
+select * from unnest(t1.a, "$.a") as f;
+select * from unnest(t1.a, "$.a", true) as f;
+select * from unnest(t1.a);
+select * from unnest(t1.a, "$.b");
+select * from unnest(t1.a, "$.a");
+select * from unnest(t1.a, "$.a", true) as f;
+
+#Test for other cases
+select col from unnest(t1.a, "$.b") as f;
+select f.col from unnest(t1.a, "$.b") as f;
+select f.seq, f.col from unnest(t1.a, "$.b") as f;
+select distinct(f.seq) from unnest(t1.a, "$") as f;
+select `index`,path from unnest(t1.a, "$") u;
+select `index`,path from unnest(t1.a, "$") u where u.`index` = 1;
+select `index`,path from unnest(t1.a, "$") u where u.`index` is null or u.path='$.b';
+select `index`,path from unnest(t1.a, "$") u where u.`index` is null and u.path='$.b';
+select * from t1,unnest(t1.a, "$") u;
+select * from t1,unnest(t1.a, "$") u where u.`index` = 1;
+select `key`,path,value from t1, unnest(t1.a, "$") u where u.`index` is null or u.path='$.b';
+select seq,col,b from t1, unnest(t1.a, "$") u where u.`index` is null;
+select count(b),b from t1, unnest(t1.a, "$") u where u.`index` is null group by b;
+insert into t1 values ('{"add":1}',1);
+select count(seq),seq from t1, unnest(t1.a, "$") u where u.`index` is null group by seq;
+select count(seq),seq from t1, unnest(t1.a, "$") u where u.`index` is null group by seq order by seq desc;
+select count(`index`),`index`,sum(`index`) from t1, unnest(t1.a, "$") u where u.`index` is not null group by `index` order by sum(`index`) desc;
+select * from t1,unnest(t1.a,"$.b",true) u where seq>1;
```

Comment:
Can we add a subquery case?   Example,

with myt as (select t1.a, from t1 where t1.b > 1) select * from unnest (myt.a, "$.b", true);

Reply:
Sure